src/query/api/v1/handler/namespace/common.go

```diff
@@ -155,3 +156,46 @@ func RegisterRoutes(
 		applyMiddleware(NewSchemaResetHandler(client, instrumentOpts).ServeHTTP, defaults))
 	r.HandleFunc(M3DBSchemaURL, schemaResetHandler.ServeHTTP).Methods(DeleteHTTPMethod)
 }
+
+func validateNamespaceAggregationOptions(mds []namespace.Metadata) error {
+	unaggCount := 0
+	resolutionRetentionMap := make(map[resolutionRetentionKey]bool, len(mds))
+
+	for _, md := range mds {
+		aggOpts := md.Options().AggregationOptions()
+		if aggOpts == nil || len(aggOpts.Aggregations()) == 0 {
+			continue
+		}
+
+		retention := md.Options().RetentionOptions().RetentionPeriod()
+		for _, agg := range aggOpts.Aggregations() {
+			if agg.Attributes.AggregationType == namespace.Unaggregated {
+				unaggCount++
+				if unaggCount > 1 {
+					return errors.New("cannot have more than one unaggregated namespace")
```

Comment:
Based on offline discussions, I believe we decided to not throw in this case so that it isn't blocked for the generic time series use-case. Later, though, we'll want to add this metric-specific validation likely from some more metrics-specific API.

Reply:
Haven't updated the PR yet, but yeah, this will be gone.